Libplanet.Net/Consensus/ConsensusContext.cs

```diff
@@ -142,7 +171,75 @@ public void Dispose()
+        public void NewHeight(long height) =>
+            NewHeight(height, null, null, null, null, null, null);
+
+        /// <summary>
+        /// Committing the block to the <see cref="BlockChain{T}"/>.
+        /// </summary>
+        /// <param name="block">A <see cref="Block{T}"/> to committing to the
+        /// <see cref="BlockChain{T}"/>.
+        /// </param>
+        /// <remarks>the method is called when a block is voted by <see cref="Context{T}"/>
+        /// in <see cref="Libplanet.Net.Consensus.Step.EndCommit"/>.
+        /// </remarks>
+        public void Commit(Block<T> block)
+        {
+            _logger.Debug("Committing block #{Index} {Block}.", block.Index, block.Hash);
+            _blockChain.Append(block);
+        }
+
+        /// <summary>
+        /// Handling the received <see cref="ConsensusMessage"/>.
+        /// </summary>
+        /// <param name="consensusMessage">a received <see cref="ConsensusMessage"/> from any
+        /// bounding validator.
+        /// </param>
+        /// <exception cref="InvalidHeightMessageException"> Thrown if the given message is lower
+        /// than current <see cref="Height"/>.
+        /// </exception>
+        public void HandleMessage(ConsensusMessage consensusMessage)
+        {
+            long height = consensusMessage.Height;
+            if (height < Height)
+            {
+                throw new InvalidHeightMessageException(
+                    $"Received message's height {height} is lower than " +
+                    $"current context's height {Height}.",
+                    consensusMessage);
+            }
+
+            if (!_contexts.ContainsKey(height))
+            {
+                _contexts[height] = new Context<T>(
+                    this,
+                    _blockChain,
+                    height,
+                    _privateKey,
+                    _getValidators(height).ToList());
+            }
+
+            _contexts[height].ProduceMessage(consensusMessage);
+        }
+
+        /// <summary>
+        /// Returns the summary for <see cref="ConsensusContext{T}"/>.
+        /// </summary>
+        /// <returns>Returns the current height <see cref="Context{T}"/>. if there's no instance of
+        /// <see cref="Context{T}"/> for current height, returns "No context".
+        /// </returns>
+        public override string ToString() => _contexts.ContainsKey(Height)
+            ? _contexts[Height].ToString()
+            : "No context";
+
+        internal void NewHeight(
+            long height,
+            EventHandler<Exception>? initialExceptionHandler,
+            EventHandler<(Step Step, TimeSpan TimeSpan)>? initialTimeoutHandler,
+            EventHandler<int>? initialTimeoutProcessHandler,
+            EventHandler<(int MessageLogSize, int Round, Step Step)>? initialStateHandler,
+            EventHandler<ConsensusMessage>? initialMessageHandler,
+            EventHandler<System.Action>? initialMutationHandler)
```

Comment:
If I understand correctly, this together with `AttachHandlers` is to use a specific handler on a `Context` with a specific height. I think it'd be better to have something like event handlers in `ConsensusContext.Event.cs` akin to `Context.Event.cs` (doesn't necessarily have to be in a separate file, just my preference) that passes on the events from all `Context`s, if possible, as a proper channel.